lib/json/common.rb

```diff
@@ -179,35 +301,148 @@ def parse!(source, opts = {})
     Parser.new(source, **(opts||{})).parse
   end
 
-  # Generate a JSON document from the Ruby data structure _obj_ and return
-  # it. _opts_ is
-  # * a Hash like object (responding to +to_hash+),
-  # * or an object convertible into a hash by a +to_h+ method,
-  # * or a <tt>JSON::State</tt> object.
-  #
-  # If hash-alike or hash-convertible object is provided, it is internally
-  # converted into a State object.
-  #
-  # The default options are set to create the shortest possible JSON text
-  # in one line, check for circular data structures and do not allow NaN,
-  # Infinity, and -Infinity.
-  #
-  # An _opts_ hash can have the following keys:
-  # * *indent*: a string used to indent levels (default: <tt>''</tt>),
-  # * *space*: a string that is put after a <tt>:</tt> pair delimiter (default: <tt>''</tt>),
-  # * *space_before*: a string that is put before a <tt>:</tt> pair delimiter (default: <tt>''</tt>),
-  # * *object_nl*: a string that is put at the end of a JSON object (default: <tt>''</tt>),
-  # * *array_nl*: a string that is put at the end of a JSON array (default: <tt>''</tt>),
-  # * *allow_nan*: true if NaN, Infinity, and -Infinity should be
-  #   generated, otherwise an exception is thrown if these values are
-  #   encountered. This options defaults to false.
-  # * *max_nesting*: The maximum depth of nesting allowed in the data
-  #   structures from which JSON is to be generated. Disable depth checking
-  #   with <tt>max_nesting: false</tt>, it defaults to 100.
-  #
-  # See also the fast_generate for the fastest creation method with the least
-  # amount of sanity checks, and the pretty_generate method for some
-  # defaults for pretty output.
+  # Argument +obj+ is the Ruby object to be converted to \JSON.
+  #
+  # Argument +opts+, if given, contains options for the generation, and must be a
+  # {Hash-convertible object}[doc/implicit_conversion_rdoc.html#label-Hash-Convertible+Objects]
+  # (implementing +to_hash+).
+  #
+  # Returns a \String containing the generated \JSON data.
+  #
+  # See also JSON.fast_generate, JSON.pretty_generate.
+  #
+  # ---
+  #
+  # When +obj+ is an
+  # {Array-convertible object}[doc/implicit_conversion_rdoc.html#label-Array-Convertible+Objects]
+  # (implementing +to_ary+), returns a \String containing a \JSON array:
+  #   obj = ["foo", 1.0, true, false, nil]
+  #   json = JSON.generate(obj)
+  #   json # => "[\"foo\",1.0,true,false,null]"
+  #   json.class # => String
```

Comment:
Remove, overly precise.

Reply:
Last line removed in next push.  Is that what you meant?  If removing all four lines, the descriptive text should also be removed?

Reply:
Yup, just last lines. Github isn't super clear for multi-line selection but when that's the case there's a comment.